flytekit/experimental/eager_function.py

```diff
@@ -605,16 +617,25 @@ def _internal_demo_remote(remote: FlyteRemote) -> FlyteRemote:
 
 def _internal_remote(
     remote: FlyteRemote,
-    client_secret_group: str,
-    client_secret_key: str,
+    client_secret_group: Optional[str],
+    client_secret_key: Optional[str],
+    client_secret_env_var: Optional[str],
 ) -> FlyteRemote:
     """Derives a FlyteRemote object from a yaml configuration file, modifying parts to make it work internally."""
-    assert client_secret_group is not None, "secret_group must be defined when using a remote cluster"
-    assert client_secret_key is not None, "secret_key must be defined a remote cluster"
     secrets_manager = current_context().secrets
     client_secret = secrets_manager.get(client_secret_group, client_secret_key)
     # get the raw output prefix from the context that's set from the pyflyte-execute entrypoint
     # (see flytekit/bin/entrypoint.py)
+
+    if client_secret_env_var is not None:
+        # this creates a remote client where the env var client secret is sufficient for authentication
+        os.environ[client_secret_env_var] = client_secret
+        try:
+            remote_cls = type(remote)
+            return remote_cls()
```

Comment:
This assumes that the other `FlyteRemote` object can be created without a `Config` and uses an environment variable. Is this okay for OSS?

Also, should this carry over the `default_domain` and `default_project` from the original remote obj? Or should the other remote object figure out the domain and project from the execution environment?

Reply:
ooo yeah it should probably carry over those args.

admittedly this code path is basically: "tell me you only run on serverless without saying you run on serverless" 🙃

Reply:
Basically `client_secret_env_var` only really applies to serverless